src/BSON/Binary.stub.php

```diff
@@ -7,7 +7,7 @@
 
 namespace MongoDB\BSON;
 
-final class Binary implements BinaryInterface, \JsonSerializable, Type, \Stringable
+class Binary implements BinaryInterface, \JsonSerializable, Type, \Stringable
```

Comment:
This classes does not define any method that is not in one of the interfaces. This means you don't need to make it extensible: the new `BinaryVector` class could implement all this interfaces (re-using the underlying C functions).

In the library, [I used `MongoDB\BSON\Binary` type in the builder](https://github.com/mongodb/mongo-php-library/blob/25dd588cdd3360294fa8539c29f1cec66dc33c56/generator/config/expressions.php#L27), but I would change it to use `BinaryInterface`. Why this interface doesn't extend `MongoDB\BSON\Type`?

Reply:
The interfaces were never very useful for PHPC. One of the original use cases was have an embedded object masquerade as a UTCDateTime and incorporate a timezone, but the benefits are mainly for userland applications.

I think sharing a single Binary class is ideal for anything that would expect the BSON type, and avoiding a separate BinaryVector child class seems like it'd make things easier for users (and simplify what comes back from BSON-to-PHP conversion).